src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,53 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
+Animal.alive = [];
+
+Animal.prototype.addToAlive = function() {
+  Animal.alive.push(this);
+};
+
+Animal.prototype.removeFromAlive = function() {
+  const index = Animal.alive.indexOf(this);
+
+  if (index !== -1) {
+    Animal.alive.splice(index, 1);
+  }
+};
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+    this.addToAlive();
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.addToAlive();
+  }
+
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden !== true) {
```

Comment:
```suggestion
    if (animal instanceof Herbivore && !animal.hidden) {
```

Same but a bit shorter